lib/src/implementation/blocs/network_filterable.dart

```diff
@@ -1,6 +1,182 @@
 import 'package:stx_bloc_base/src/base/index.dart';
 import 'package:stx_bloc_base/src/implementation/index.dart';
 
+/// A utility class that simplifies working with asynchronous data, specifically for filtering. Instead of manually adding the event, simply call the desired method, which internally adds the corresponding event to the [Bloc] and ensures it is handled by internally.
+///
+/// Example usage:
+///
+/// ```dart
+/// class Note {
+///   Note(this.id, this.message);
+///
+///   final int id;
+///   final String message;
+/// }
+///
+/// enum Filter {
+///   even,
+///   odd;
+///
+///   bool get isEven => this == Filter.even;
+///   bool get isOdd => this == Filter.odd;
+/// }
+///
+/// typedef MyData = List<Note>;
+/// // Here `Filter` is used to filter the data by `id`. Any type can be used here, for example, `bool`, `int`, `String`, enum, etc.
+/// typedef MyState = NetworkFilterableState<MyData, Filter>;
+///
+/// // The state contains the `data` of any type in this case `List<Note>` and `status` (NetworkStatus), which is by default [NetworkStatus.initial], `visibleData` will be used to display the data in the UI based on the user's search input, and `query` is used to store the user's search input and `filter` is used to store the user's filter input.
```

Comment:
Please simplify . No one wants to read full implementation in class documentation